### PR TITLE
Fix/intial preferences

### PR DESCRIPTION
**Pull request by felipe-najson-ntf**
InitialPreferences working properly. Related to issues #121  #122

### CODE REVIEW

**Comment by borisrorsvort** [comment]
@felipe-najson-ntf any idea when this will be merged?

**Comment by felipe-najson-ntf** [comment]
Hello again @borisrorsvort, I'm waiting for the approval by @pooyaj. We hope these days to merge this pr and publish the new version of consent manager.

File: src/consent-manager-builder/index.tsx
Comment (pooyaj) [inline src/consent-manager-builder/index.tsx:231]: @felipe-najson-ntf This line could be problematic. We want to load `analytics.js` even if the user doesn't provide the `initialPreferences`. I think adding this `if` clause will lead to cases where users already consented to everything, and their consent settings is stored in cookies, but then the `analytics.js` doesn't load.

Comment (pooyaj) [reply src/consent-manager-builder/index.tsx:231]: This is the work in progress branch: 
Storybook

Comment (pooyaj) [reply src/consent-manager-builder/index.tsx:231]: This is the master branch: 
Storybook

File: src/consent-manager-builder/index.tsx
Comment (pooyaj) [inline src/consent-manager-builder/index.tsx:320]: Removing this `if` is not ideal as well. To test the difference: 
Work-in-progress branch: When opening the consent modal and clicking `save` the page will refresh even if we haven't changed anything. 
Master branch: When opening the consent modal and clicking `save` the page won't refresh unless there is a change.

**Review by pooyaj — CHANGES_REQUESTED** [review]
@felipe-najson-ntf Sorry for the delay in reviewing the PR. There are some issues that we need to solve to make sure there is no regression. I pointed to two issues in the review. Also I suggest adding a new StoryBoard page that covers the two issues that this PR is trying fix, so we can test the changes easier.

File: src/consent-manager-builder/index.tsx
Comment (felipe-najson-ntf) [reply src/consent-manager-builder/index.tsx:231]: Good morning @pooyaj, I do not agree with you on this point. I think that is exactly what users are complaining about, analytics.js initializes automatically without user consent. The idea is to load the initial preferences to the cookie "tracking-preferences" but NOT initialize. 

With this we tried not to affect other behaviors of the library except when there are Initial preferences, do not execute it. With this solution when loading the page we will have:
image

Once the consent is accepted in the modal, tracking starts:
image

**_Note: I have changed the if sentence to something more precise in the following commit._**

Comment (felipe-najson-ntf) [reply src/consent-manager-builder/index.tsx]: @pooyaj I just left a pr with the new solution. If you still have doubts of why I included the if sentence we can have a small meeting to conclude with this issue. Greetings

File: src/consent-manager-builder/index.tsx
Comment (felipe-najson-ntf) [reply src/consent-manager-builder/index.tsx:320]: I agree with you, I did not realize that the page was refreshing when removing the if. This was fixed in the last commit.

**Review by davidvenin — APPROVED** [review]

File: src/consent-manager-builder/index.tsx
Comment (pooyaj) [reply src/consent-manager-builder/index.tsx]: Sorry for another long delay for the response. I think there is a misunderstanding about the purpose of the `initialPreferences`. When we set the `initialPreferences`, it means that we are **automatically opt-ing in** users to those preferences only if there is no prior consent from the user. It means we assume consent from the user according to the `initialPreferences`. 

Another important point here is that if there is already a consent ( in the cookies, or from the initial preferences ) we want to attempt to call `conditionallyLoadAnalytics`. If we don't do that, user ends up in a situation where they have to consent everytime they refresh the page ( or every time the consent manager component mounts )

File: src/consent-manager-builder/index.tsx
Comment (pooyaj) [inline src/consent-manager-builder/index.tsx:253]: We don't need this here. If we set this to true, if will impact the `handleSaveConsent` and leads to unnecessary refresh.

File: src/consent-manager-builder/index.tsx
Comment (pooyaj) [inline src/consent-manager-builder/index.tsx:326]: We have to update the `handleSaveConsent` function signature to allow for `boolean` ... so something like this: 
```
handleSaveConsent = (newPreferences: CategoryPreferences | undefined | boolean, shouldReload: boolean)
```

**Comment by borisrorsvort** [comment]
@felipe-najson-ntf any news?

**Comment by felipe-najson-ntf** [comment]
> @felipe-najson-ntf any news?

Hello again @borisrorsvort. I just saw the requests from @pooyaj and we will have a meeting next week to close this issue since we are not understanding each other well. 
Sorry for the delay of this pr, I hope that next week the new version will be published

File: src/consent-manager-builder/index.tsx
Comment (felipe-najson-ntf) [reply src/consent-manager-builder/index.tsx:326]: Agree with you.

**Comment by borisrorsvort** [comment]
@felipe-najson-ntf  @pooyaj Thanks! I really urge you to solve this matter asap as it basically makes all user of that component outlaw in Europe gdpr wise, and that issue has been open for more than a year.

**Review by pooyaj — APPROVED** [review]

**Pull request merged**
